File: ЛР6/ЛР6/MFCLibrary1/Starosta_Stoler.cpp
#include "pch.h"
#include "Starosta_Stoler.h"
using namespace std;

IMPLEMENT_SERIAL(Starosta_Stoler, Student_Stoler, 1 | VERSIONABLE_SCHEMA)

Starosta_Stoler::Starosta_Stoler() :rating(), phone_number() {}

Starosta_Stoler::~Starosta_Stoler() {}

void Starosta_Stoler::Serialize(CArchive& ar)
{
	Student_Stoler::Serialize(ar);
	if (ar.IsStoring()) {
		ar << rating;
		ar << phone_number;
	}
	else {
		ar >> rating;
		ar >> phone_number;
	}
}

File: ЛР5/Stoler_LR5/Starosta_Stoler.cs
using System;

namespace Stoler_LR5
{
    [Serializable]
    public class Starosta_Stoler : Student_Stoler
    {
        public double Rating { get; set; }
        public long Phone_number { get; set; }
        public override void PrintToConsole()
        {
            base.PrintToConsole();
            Console.WriteLine("Рейтинг: " + Rating + "\n" + "Номер телефона: " + Phone_number);
        }
        public override void Read()
        {
            base.Read();
            Console.WriteLine("Введите рейтинг старосты");
            Rating = Convert.ToDouble(Console.ReadLine());
            Console.WriteLine("Введите номер телефона старосты");
            Phone_number = (long)Convert.ToDouble(Console.ReadLine());
        }
    }
}
